attack.go

```diff
@@ -211,6 +202,93 @@ func attack(opts *attackOpts) (err error) {
 	}
 }
 
+func newHitter(opts *attackOpts) (vegeta.Hitter, error) {
+	tlsc, err := tlsConfig(opts.insecure, opts.certf, opts.keyf, opts.rootCerts)
+	if err != nil {
+		return nil, err
+	}
+
+	if opts.http2 || opts.h2c || opts.unixSocket != "" {
+		dialer := net.Dialer{
+			LocalAddr: &net.TCPAddr{
+				IP: opts.laddr.IP,
+				Zone: opts.laddr.Zone,
+			},
+			KeepAlive: 30 * time.Second,
+		}
+
+		tr := &http.Transport{
+			Proxy:               http.ProxyFromEnvironment,
+			DialContext:         dialer.DialContext,
+			TLSClientConfig:     tlsc,
+			MaxIdleConnsPerHost: opts.connections,
+			MaxConnsPerHost:     opts.maxConnections,
+			DisableKeepAlives: !opts.keepalive,
+			ProxyConnectHeader: opts.proxyHeaders.Header,
+		}
+
+		client := &http.Client{
+			Timeout: opts.timeout,
+			Transport: tr,
+			CheckRedirect: func(_ *http.Request, via []*http.Request) error {
+				switch {
+				case opts.redirects == vegeta.NoFollow:
+					return http.ErrUseLastResponse
+				case opts.redirects < len(via):
+					return fmt.Errorf("stopped after %d redirects", opts.redirects)
+				default:
+					return nil
+				}
+			},
+		}
+
+		switch {
+		case opts.http2:
+			if err = http2.ConfigureTransport(tr); err != nil {
+				return nil, err
+			}
+		case opts.h2c:
+			client.Transport = &http2.Transport{
+				AllowHTTP: true,
+				DialTLS: func(network, addr string, cfg *tls.Config) (net.Conn, error) {
+					return tr.DialContext(context.Background(), network, addr)
+				},
+			}
+		case opts.unixSocket != "":
+			tr.DialContext = func(_ context.Context, _, _ string) (net.Conn, error) {
+				return net.Dial("unix", opts.unixSocket)
+			}
+		}
+
+		return &vegeta.NetHTTPHitter{
+			Chunked: opts.chunked,
+			MaxBody: opts.maxBody,
+			Client:  client,
+		}, nil
+	}
+
+	// TODO(tsenart): Add fasthttp support for:
+	//  - opts.redirects
+	//  - opts.connections (max idle conns per host)
+	//  - opts.keepalive
```

Comment:
Seems like keep-alives can be configured either through headers and/or these durations: https://github.com/valyala/fasthttp/blob/master/client.go#L570